src/Authoring/WinRT.SourceGenerator/DiagnosticHelpers.cs

```diff
@@ -368,97 +367,58 @@ private string SimplifySyntaxTypeString(string syntaxType)
-            { "System.Numerics.Plane", new List<string> { "D", "Normal" } },
-            { "System.Numerics.Quaternion", new List<string> { "Identity" ,"IsIdentity" } },
-            { "System.Numerics.Vector2", new List<string> { "One", "UnitX", "UnitY", "Zero" } },
-            { "System.Numerics.Vector3", new List<string> { "One", "UnitX", "UnitY", "UnitZ", "Zero" } },
-            { "System.Numerics.Vector4", new List<string> { "One", "UnitW", "UnitX", "UnitY", "UnitZ", "Zero" } },
-            { "System.Type", new List<string> { "Assembly", "AssemblyQualifiedName", "Attributes", "BaseType", "ContainsGenericParamters", "DeclaringMethod", "DeclaringType", "DefaultBinder",
-                "FullName", "GenericParameterAttributes", "GenericParameterPosition", "GenericTypeArguments", "GUID", "HasElementType", "IsAbstract",
-                "IsAnsiClass", "IsArray", "IsAutoClass", "IsAutoLayout", "IsByRef", "IsByRefLike", "IsClass", "IsCOMObject",
-                "IsConstructedGenericType", "IsContextful", "IsEnum", "IsExplicitLayout", "IsGenericMethodParameter", "IsGenericParameter", "IsGenericType", "IsGenericTypeDefinition",
-                "IsGenericTypeParameter", "IsImport", "IsInterface", "IsLayoutSequential", "IsMarshalByRef", "IsNested", "IsNestedAssembly", "IsNestedFamANDAssem", "IsNestedFamily",
-                "IsNestedFamORAssem", "IsNestedPrivate", "IsNestedPublic", "IsNotPublic", "IsPointer", "IsPrimitive", "IsPublic", "IsSealed", "IsSecurityCritical", "IsSecuritySafeCritical",
-                "IsSecurityTransparent", "IsSerializable", "IsSignatureType", "IsSpecialName", "IsSZArray", "IsTypeDefinition", "IsUnicodeClass", "IsValueType", "IsVariableBoundArray",
-                "IsVisible", "MemberType", "Module", "Namespace", "ReflectedType", "StructLayoutAttribute", "TypeHandle", "TypeInitializer", "UnderlyingSystemType"} },
-            { "System.Collections.Generic.IEnumerator`1", new List<string> { "Current" } },
-            { "System.Collections.Generic.KeyValuePair`2", new List<string> { "Key", "Value" } },
-            { "System.Collections.Generic.IReadOnlyDictionary`2", new List<string> { "Keys", "Values", "Count" } },
-            { "System.Collections.Generic.IDictionary`2", new List<string> { "Keys", "Values", "Count" } },
-            { "System.Collections.Generic.IList`1", new List<string> { "IsFixedSize", "IsReadOnly", "Item" } },
-            { "System.Collections.IList`1", new List<string> { "IsFixedSize", "IsReadOnly", "Item" } }
+            "System.Collections.Generic.ICollection`1",
+            "System.Collections.Generic.IReadOnlyCollection`1",
+            "System.Collections.IEnumerable",
+            "System.Collections.IEnumerator",
+            "System.Collections.IList"
         };
 
-        private static readonly string[] nonWindowsRuntimeInterfaces = 
+        private static readonly HashSet<string> nonWindowsRuntimeInterfaces = new HashSet<string>()
         {

```

Comment:
Should the other 4 types here be qualified names similar to the first one?

Reply:
Well gettypebymetadataname fails for the qualified ones, and I checked one of our tests for these interfaces, and when the class implements it, the interface's containing namespace is global (unless qualified when giving the class the base type), which means qualified names won't help, just metadata name.

Reply:
My only concern was if someone used the full typename or somewhere whether we will miss it.  But if we are comparing metadata name, it probably is fine. Just seemed inconsistent with System.Exception.  And what happens if someone had their own version of IAsyncAction in their own namespace?

Reply:
Yeah we do catch full name, but I bet we won't let it happen if they write IAsyncAction, but should they be allowed to write IAsyncAction? they are are encouraged (in windmdexp) to use IAsyncInfo class for exception code

Reply:
Well I meant IAsyncAction from some other namespace (different type).

Reply:
I meant is Exception a valid winrt type? implementing IAsyncInfo would mean having a property that is an Exception.. Maybe there's an edge case there with them also writing their own Exception class?

Reply:
I think it is valid, but probably want to confirm.  You probably also want to check if inheriting is allowed, but I think not.